jobs/pacman-awsrules/src/test/java/com/tmobile/cloud/awsrules/cloudtrail/DenyPublicAccessS3BucketStoresCloudTrailLogsTest.java

```diff
@@ -0,0 +1,68 @@
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({PacmanUtils.class, Annotation.class})
+public class DenyPublicAccessS3BucketStoresCloudTrailLogsTest {
+
+    @InjectMocks
+    DenyPublicAccessS3BucketStoresCloudTrailLogs denyPublicAccessS3BucketStoresCloudTrailLogs;
+
+    @Test
+    public void executeTest() throws Exception{
+        mockStatic(Annotation.class);
+        when(Annotation.buildAnnotation(anyObject(),anyObject())).thenReturn(CommonTestUtils.getMockAnnotation());
+        mockStatic(PacmanUtils.class);
+        when(PacmanUtils.doesAllHaveValue(anyString(),anyString())).thenReturn(
+                true);
+        assertThat(denyPublicAccessS3BucketStoresCloudTrailLogs.execute(getMapString("r_123 "),getMapString("r_123 ")), is(notNullValue()));
+        when(PacmanUtils.doesAllHaveValue(anyString(),anyString())).thenReturn(
+                false);
+        assertThatThrownBy(
+                () -> denyPublicAccessS3BucketStoresCloudTrailLogs.execute(getMapString("r_123 "),getMapString("r_123 "))).isInstanceOf(InvalidInputException.class);
+
+    }
+
+
+    @Test
+    public void getHelpTextTest(){
+        assertThat(denyPublicAccessS3BucketStoresCloudTrailLogs.getHelpText(), is(notNullValue()));
+    }
+
+    public static Map<String, String> getMapString(String passRuleResourceId) {
+        Map<String, String> commonMap = new HashMap<>();
+        commonMap.put("executionId", "1234");
+        commonMap.put("_resourceid", passRuleResourceId);
+        commonMap.put("severity", "critical");
+        commonMap.put("ruleCategory", "security");
+        commonMap.put("type", "Task");
+        commonMap.put("accountid", "12345");
+        commonMap.put("ruleId", "deny_public_access_to_s3_bucket_used_to_store_cloudtrail_logs");
+        commonMap.put("policyId","deny_public_access_to_s3_bucket_used_to_store_cloudtrail_logs");
+        commonMap.put("policyVersion", "version-1");
+        String bucketPolicyString = "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Sid\":\"AWSCloudTrailAclCheck20150319\",\"Effect\":\"Allow\",\"Principal\":{\"Service\":\"cloudtrail.amazonaws.com\"},\"Action\":\"s3:GetBucketAcl\",\"Resource\":\"arn:aws:s3:::aws-cloudtrail-logs-500559730414-4d2373d6\",\"Condition\":{\"StringEquals\":{\"AWS:SourceArn\":\"arn:aws:cloudtrail:us-east-1:500559730414:trail/management-events\"}}},{\"Sid\":\"AWSCloudTrailWrite20150319\",\"Effect\":\"Allow\",\"Principal\":{\"Service\":\"cloudtrail.amazonaws.com\"},\"Action\":\"s3:PutObject\",\"Resource\":\"arn:aws:s3:::aws-cloudtrail-logs-500559730414-4d2373d6/AWSLogs/500559730414/*\",\"Condition\":{\"StringEquals\":{\"AWS:SourceArn\":\"arn:aws:cloudtrail:us-east-1:500559730414:trail/management-events\",\"s3:x-amz-acl\":\"bucket-owner-full-control\"}}}]}";
```

Comment:
1. can you replace real data (like account ID for instance) with fake data?
2. is it possible to reuse later for similar tests? If so then let's make a common static method for tests which return this string. of course, only if you think it may be reused later, other wise - keep it as it is.

Reply:
replaced real data with fake data